contracts/interchain-token-service/src/contract.rs

```diff
@@ -526,8 +510,57 @@ impl InterchainTokenService {
                     );
                 }
             }
-            Message::DeployInterchainToken(_) => {
-                todo!()
+            Message::DeployInterchainToken(DeployInterchainToken {
+                token_id,
+                name,
+                symbol,
+                decimals,
+                minter,
+            }) => {
+                ensure!(
+                    Self::token_id_config(env, token_id.clone()).is_err(),
+                    ContractError::TokenAlreadyDeployed
+                );
+
+                let token_metadata = TokenMetadata {
+                    name,
+                    symbol,
+                    decimal: decimals as u32,
+                };
+
+                ensure!(
+                    validate_token_metadata(token_metadata.clone()).is_ok(),
+                    ContractError::InvalidTokenMetaData
+                );
+
+                let minter_address = if let Some(minter) = minter {
+                    let addr = Address::from_xdr(env, &minter).expect("address conversion failed");
+
+                    ensure!(
```

Comment:
Do we have this check on the receiving side for EVM ITS?

Reply:
I don't see it in the EVM contracts. I took it from the `deploy_interchain_token` in the Stellar implementation: https://github.com/axelarnetwork/axelar-cgp-soroban/blob/8430b939fe3a330161ee7318218a85ee6721254d/contracts/interchain-token-service/src/contract.rs#L189.

Seemed to make sense. Should I remove?